### PR TITLE
remove dependencies not reqd for 1.15+

### PR DESCRIPTION
**Pull request by aramase**
Fixes https://github.com/deislabs/secrets-store-csi-driver/issues/104

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,6 +44,10 @@ The diagram below illustrates how Secrets Store CSI Volume works.
 
 ### Prerequisites
 
+#### Supported kubernetes versions
+
+secrets-store-csi-driver is supported only for cluster versions >= v1.15.0
```
Comment (ritazh) [inline README.md]: nit: `v1.15.0+`

File: README.md [hunk 2]
```diff
@@ -86,58 +90,48 @@ Expected output:
 
 ```console
 NAME:   csi-secrets-store
+LAST DEPLOYED: Mon Nov  4 12:51:55 2019
```
Comment (ritazh) [inline README.md]: let's avoid dates here as it makes the readme look out of date

File: README.md [hunk 3]
```diff
@@ -86,58 +90,48 @@ Expected output:
...
-
-NAME                                              READY  STATUS             RESTARTS  AGE
-csi-secrets-store-attacher-0                      0/1    ContainerCreating  0         2s
-csi-secrets-store-secrets-store-csi-driver-q74wf  0/2    ContainerCreating  0         2s
-csi-secrets-store-secrets-store-csi-driver-ssw6r  0/2    ContainerCreating  0         2s
-csi-secrets-store-provider-azure-pksfd            0/2    ContainerCreating  0         2s
-csi-secrets-store-provider-azure-sxht2            0/2    ContainerCreating  0         2s
+NAME                              DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR                AGE
+csi-secrets-store-provider-azure  2        2        0      2           0          beta.kubernetes.io/os=linux  1s
+csi-secrets-store-provider-vault  2        2        0      2           0          <none>                       1s
```
Comment (ritazh) [inline README.md]: Can you pls also update the `[ALTERNATIVE DEPLOYMENT OPTION] Using Deployment Yamls` section to remove the unnecessary yamls.

File: charts/secrets-store-csi-driver/templates/role_binding.yaml [hunk 4]
```diff
@@ -12,6 +12,6 @@ subjects:
   name: default
```
Comment (ritazh) [inline charts/secrets-store-csi-driver/templates/role_binding.yaml]: I think we can remove this.

File: deploy/rbac-secretproviderclass.yaml [hunk 5]
```diff
@@ -11,7 +17,7 @@ subjects:
   name: default
```
Comment (ritazh) [inline deploy/rbac-secretproviderclass.yaml]: Same. I think we can remove this.

File: README.md (hunk 1 above)
Comment (aramase) [reply README.md]: done!

File: README.md (hunk 2 above)
Comment (aramase) [reply README.md]: done!

File: README.md (hunk 3 above)
Comment (aramase) [reply README.md]: done!

File: deploy/rbac-secretproviderclass.yaml (hunk 5 above)
Comment (aramase) [reply deploy/rbac-secretproviderclass.yaml]: done!

File: charts/secrets-store-csi-driver/templates/role_binding.yaml (hunk 4 above)
Comment (aramase) [reply charts/secrets-store-csi-driver/templates/role_binding.yaml]: done!

File: README.md [hunk 6]
```diff
@@ -90,54 +94,43 @@ NAMESPACE: dev
...
```
Comment (ritazh) [inline README.md]: We should add the following in https://github.com/deislabs/secrets-store-csi-driver/blob/master/deploy/provider-vault.yaml similar to the azure provider:

```yaml
nodeSelector:
  beta.kubernetes.io/os: linux
```

@malnick @kalafut FYI

Comment (aramase) [reply README.md]: done!

**Comment by ritazh** [comment]
One last thing, can you pls add the following to Chart.yaml

```yaml
kubeVersion: ">=1.15.0-0"
```

**Pull request merged**